<a href="https://colab.research.google.com/github/MounicaK12/Python-Project/blob/main/Mounica_Kodavalla_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Analysis of the Titanic sink survivors

Here I used pandas and numpy

In [1]:
import pandas as pd
import numpy as np

Below is the code for extracting the data from csv file
pandas dataframe is for converting file to dataframe
read_csv function is used to read the csv file

In [4]:
df= pd.DataFrame(pd.read_csv('/content/train (2).csv'))
df.head()

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0            1         0       3  ...   7.2500   NaN         S
1            2         1       1  ...  71.2833   C85         C
2            3         1       3  ...   7.9250   NaN         S
3            4         1       1  ...  53.1000  C123         S
4            5         0       3  ...   8.0500   NaN         S

[5 rows x 12 columns]

In [5]:
df.shape

(891, 12)

There are 891 passengers and 12 sets informations of each of them.

Now, lets check the null values

In [6]:
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Cabin          687
Embarked         2
dtype: int64

177 survivors did not disclose their age ,and 687 did not disclose their cabin code and 2 did not disclose the port of embarkation.

In [7]:
drop_col = df.isnull().sum()[df.isnull().sum()>(50/100 * df.shape[0])]
drop_col

Cabin    687
dtype: int64

In the cabin coloum, more than 50% of the data is missing.

As a large amount of data for this column is missing, we can remove it from the dataset.

In [8]:
drop_col.index

Index(['Cabin'], dtype='object')

In [9]:
df.drop(drop_col.index, axis=1, inplace= True )
df.isnull().sum()

PassengerId      0
Survived         0
Pclass           0
Name             0
Sex              0
Age            177
SibSp            0
Parch            0
Ticket           0
Fare             0
Embarked         2
dtype: int64

As seen above, the cabin column is dropped.

Now, as age is important, lets fill in the empty values with the age of most of the passengers i.e., mode.

In [13]:
df.fillna(df.mode(),inplace = True)
df.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       2
dtype: int64

Now, the age empty values have been filled 

Now lets fill in the 2 empty embarked port values.

In [14]:
df['Embarked'].describe()

count     889
unique      3
top         S
freq      644
Name: Embarked, dtype: object

In [17]:
df['Embarked'].fillna('S',inplace = True)
df.isnull().sum()

PassengerId    0
Survived       0
Pclass         0
Name           0
Sex            0
Age            0
SibSp          0
Parch          0
Ticket         0
Fare           0
Embarked       0
dtype: int64

Empty embarket port values have been replaced with the most commonly occured port 'S' Southampton.

Now, there are no empty values

Lets find the correlation between the values

In [18]:
df.corr()

PassengerId  Survived    Pclass  ...     SibSp     Parch      Fare
PassengerId     1.000000 -0.005007 -0.035144  ... -0.057527 -0.001652  0.012658
Survived       -0.005007  1.000000 -0.338481  ... -0.035322  0.081629  0.257307
Pclass         -0.035144 -0.338481  1.000000  ...  0.083081  0.018443 -0.549500
Age             0.033207 -0.069809 -0.331339  ... -0.232625 -0.179191  0.091566
SibSp          -0.057527 -0.035322  0.083081  ...  1.000000  0.414838  0.159651
Parch          -0.001652  0.081629  0.018443  ...  0.414838  1.000000  0.216225
Fare            0.012658  0.257307 -0.549500  ...  0.159651  0.216225  1.000000

[7 rows x 7 columns]

There is a strong negative correlation between the fare and the passenger class. As the fare increased(70 pounds approx), the passenger got a better class(1). lower the fare(around 10 pounds) they got a worse class(3).
The age and survival had almost no correlation. 

In [45]:
df['Sex']= [0 if df['Sex'][i]=='male' else 1 for i in df.index]
df.groupby(['Sex'])['Survived'].mean()

Sex
0    0.188908
1    0.742038
Name: Survived, dtype: float64

Here we can see that more female passengers survived the drown than the male passengers. It makes sense as women and kids were evacuated from the ship.

In [46]:
df.groupby(['Embarked'])['Survived'].mean()


Embarked
C    0.553571
Q    0.389610
S    0.339009
Name: Survived, dtype: float64

More passengers who embarked at the Cherbourg port survived compared to those who boarded on Queenstown and Southampton ports.

*Conclusion*


There are 891 passengers and 12 sets informations of each of them.

177 survivors did not disclose their age ,and 687 did not disclose their cabin code and 2 did not disclose the port of embarkation.

There is a strong negative correlation between the fare and the passenger class.

More female passengers survived the drown than the male passengers.

Passengers who embarked at the Cherbourg port survived comparitively. 